infra/modules/network/main.tf

```diff
@@ -14,13 +14,16 @@ module "aws_vpc" {
   azs  = local.availability_zones
   cidr = local.vpc_cidr
 
+  # Database subnet is only created if a database is needed
+  create_database_subnet_group = var.has_database
+
   public_subnets             = ["10.0.10.0/24", "10.0.11.0/24", "10.0.12.0/24"]
   private_subnets            = ["10.0.0.0/24", "10.0.1.0/24", "10.0.2.0/24"]
-  database_subnets           = ["10.0.5.0/24", "10.0.6.0/24", "10.0.7.0/24"]
+  database_subnets           = var.has_database ? ["10.0.5.0/24", "10.0.6.0/24", "10.0.7.0/24"] : []
   public_subnet_tags         = { subnet_type = "public" }
   private_subnet_tags        = { subnet_type = "private" }
-  database_subnet_tags       = { subnet_type = "database" }
-  database_subnet_group_name = var.database_subnet_group_name
+  database_subnet_tags       = var.has_database ? { subnet_type = "database" } : {}
+  database_subnet_group_name = var.has_database ? var.database_subnet_group_name : ""
```

Comment:
nit: I believe we don't actually need the conditionals here:

1. the tags only apply to created db subnets so if we don't create db subnets then the tags don't apply
2. similarly the subnet group name doesn't do anything if create database subnet group is false

Reply:
While they aren't strictly needed, I do prefer that the usage is explicit, rather than implied. Whenever I am tracing broken code things can become confusing if people are setting what appear to be "dead" variables.  Alternatively a comment may be helpful if preferred?

Reply:
Note: This is my personal github. I'll get signed into my usual one and make sure I'm on the right account in the future.

Reply:
@lorenyu Good catch. I've pushed a change that removes the unnecessary conditionals and adds some more commenting. The comments were a little unwieldy, so I re-organized a little to group related items together.